File: src/dataset/cpp/rna-transcription_exercism-cmccandless/claude/ClaudeSonnet4_rna_transcription_v1.cpp
#include "rna_transcription.h"

char transcription::to_rna(char ch)
{
	switch (ch)
	{
	case 'C': return 'G';
	case 'G': return 'C';
	case 'A': return 'U';
	case 'T': return 'A';
	}
	return '\0';
}

std::string transcription::to_rna(std::string s)
{
	std::string result;
	result.reserve(s.size());
	for (char ch : s) {
		result += to_rna(ch);
	}
	return result;
}

File: src/dataset/Java/rna-transcription_exercism-java-blogscot/claude/ClaudeSonnet4_rna-transcription_v1.java
class RnaTranscription {

  private static final char[] TRANSCRIPTION_MAP = new char[256];
  
  static {
    TRANSCRIPTION_MAP['C'] = 'G';
    TRANSCRIPTION_MAP['G'] = 'C';
    TRANSCRIPTION_MAP['T'] = 'A';
    TRANSCRIPTION_MAP['A'] = 'U';
  }

  String transcribe(String dnaStrand) {
    if (dnaStrand.isEmpty()) {
      return "";
    }
    
    char[] result = new char[dnaStrand.length()];
    for (int i = 0; i < dnaStrand.length(); i++) {
      result[i] = TRANSCRIPTION_MAP[dnaStrand.charAt(i)];
    }
    return new String(result);
  }

  private String transcribeRNA(String rna) {
    if (rna.length() == 1) {
      return String.valueOf(TRANSCRIPTION_MAP[rna.charAt(0)]);
    }
    return "";
  }
}